### ISSUE TITLE
pfc.txt

### ISSUE THREAD

**Issue by qizhouzhang** [alibaba-edu/High-Precision-Congestion-Control#14]
Hi yuliang. I have try three very large flow  have same dst, but the pfc.txt is empty? cc is DCQCN? Is it normal?

**Comment by liyuliang001** [comment]
How about try more flows, say 40?

**Comment by qizhouzhang** [comment]
> when I try flow number is 100, the program was terminated with this error "Command ['/home/gdp/hpcc/simulation/build/scratch/third', 'mix/config.txt'] terminated with signal SIGSEGV", Have you met before？
And when flow is number is 50, the pfc.txt is still empty

**Comment by qizhouzhang** [comment]
I understand why SIGSEGV. I have try 255 flow-> 1 dst。PFC.txt is empty。。。

**Comment by liyuliang001** [comment]
How about try analysis/trace_reader? That will tell you all packet-level
events, including the queue length when each packet arrive at each hop. So
you can see what happened.

**Comment by qizhouzhang** [comment]
Let me see see. Thanks!

**Comment by qizhouzhang** [comment]
I have tried the (DCQCN + enable win), the PFC is not triggered. But only with **DCQCN,**  the PFC is triggered. Is it reasonable?👀

**Comment by liyuliang001** [comment]
Yes. This is expected :) You can refer to HPCC paper, and you will find the
same conclusion.

**Comment by qizhouzhang** [comment]
thanks

**Comment by qizhouzhang** [comment]
Hi ! I have another question. In traffic_gen.py has param t and in config.txt also have the simulation stop time. The param t should equal the simu stop time in config.txt?

**Comment by liyuliang001** [comment]
Hi, These 2 parameters are independent. If simulation stop time is earlier
than t, simulations will stop without running all flows (since some flows
are indicated to start after simulation stop). Normally in my setting, t is
smaller than simulation stop time, to guarantee all traffic that I want is
generated.

**Comment by qizhouzhang** [comment]
I run a simulation about 10W flows. It take about 5 h. If I use traffic_gen.py generate flows with 320hosts at load 0.7, flow are 1600W. It seems to need 5 * 160 = 800h / 24 = 34d. Is there any way to speed up this process?😂

**Comment by liyuliang001** [comment]
Yes, simulation is slow :) In my experience, running 320 hosts, at load 0.3
(or 0.5?), it takes 1 day to simulate 0.1s~0.2s.

But your number seems too slow to me. What's the t in your setting?

**Comment by qizhouzhang** [comment]
I set t to 5 second. I will try smaller t👌😂 thanak you.

**Comment by qizhouzhang** [comment]
Hi Yuliang!
I want to know how to calculate the real-time throughput.Suppose a single flow's path is 1(host)->301(switch)->2(Host)?

**Comment by liyuliang001** [comment]
I think you can capture the packet trace, and then calculate the throughput yourself. That's what I did :)

To capture packet trace, you can check https://github.com/alibaba-edu/High-Precision-Congestion-Control/blob/master/simulation/run.py#L76
With this option set, the simulation also output a trace file to: https://github.com/alibaba-edu/High-Precision-Congestion-Control/blob/master/simulation/run.py#L13

Then you can use https://github.com/alibaba-edu/High-Precision-Congestion-Control/blob/master/analysis/trace_reader.cpp to read the trace file.

**Comment by songhexiang** [comment]
> I have tried the (DCQCN + enable win), the PFC is not triggered. But only with **DCQCN,** the PFC is triggered. Is it reasonable?👀

So what is the test case when PFC is triggered, what is the scale of the test? I have tried websearch 30%load for 1 second and only with DCQCN, but the pfc.txt is empty.

**Comment by zqz-net** [comment]
You can try multi to one traffic pattern.  (Multi source hosts to one destination host at the same time)

**Comment by SOULOFCINDERS** [comment]
> > when I try flow number is 100, the program was terminated with this error "Command ['/home/gdp/hpcc/simulation/build/scratch/third', 'mix/config.txt'] terminated with signal SIGSEGV", Have you met before？
> > And when flow is number is 50, the pfc.txt is still empty

Hello,I also meet the error "Command ['/home/gdp/hpcc/simulation/build/scratch/third', 'mix/config.txt'] terminated with signal SIGSEGV",have you got the reason of the error?

**Comment by 0-polarstar** [comment]
> > 我试过（DCQCN + enable win），PFC没有触发。但只有**DCQCN**才会触发 PFC。合理吗？👀
> 
> 那么PFC触发时的测试用例是什么，测试的规模是多少？我已经尝试了 websearch 30%load 1 秒，并且只使用 DCQCN，但是 pfc.txt 是空的。

我也想问，试了好几组数据都无法触发pfc

**Comment by liyuliang001** [comment]
Yes, you can check the paper Figure 11.

**Comment by zqz-net** [comment]
> > > 我试过（DCQCN + enable win），PFC没有触发。但只有**DCQCN**才会触发 PFC。合理吗？👀
> > 
> > 
> > 那么PFC触发时的测试用例是什么，测试的规模是多少？我已经尝试了 websearch 30%load 1 秒，并且只使用 DCQCN，但是 pfc.txt 是空的。
> 
> 我也想问，试了好几组数据都无法触发pfc

You can tune the pfc threshold to trigger PFC in (switch_mmu.cc file)

**Comment by 0-polarstar** [comment]
> > > > 我触发（DCQCN + win），PFC触发但只有**DCQCN**不会触发PFC。合理吗？👀
> > > 
> > > 
> > > PFC触发时的测试用例是什么，测试的规模是多少？
> > 
> > 
> > 我想也问，试了好几组数据都无法触发pfc
> 
> 您可以调整 pfc 阈值以在（switch_mmu.cc 文件）中触发 PFC

可以加个联系方式吗？我对怎么调节阈值有点不太确定

> > > > 我触发（DCQCN + win），PFC触发但只有**DCQCN**不会触发PFC。合理吗？👀
> > > 
> > > 
> > > PFC触发时的测试用例是什么，测试的规模是多少？
> > 
> > 
> > 我想也问，试了好几组数据都无法触发pfc
> 
> 您可以调整 pfc 阈值以在（switch_mmu.cc 文件）中触发 PFC

可以加您的联系方式吗？我对阈值设置不是很确定

**Comment by 0-polarstar** [comment]
可以加您联系方式吗？我对阈值设置不是很确定




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alibaba-edu/High-Precision-Congestion-Control"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月2日(星期一) 上午8:29
***@***.***&gt;;
抄送:&nbsp;"Last of the ***@***.******@***.***&gt;;
主题:&nbsp;Re: [alibaba-edu/High-Precision-Congestion-Control] pfc.txt (#14)





    
我试过（DCQCN + enable win），PFC没有触发。但只有DCQCN才会触发 PFC。合理吗？👀
  
那么PFC触发时的测试用例是什么，测试的规模是多少？我已经尝试了 websearch 30%load 1 秒，并且只使用 DCQCN，但是 pfc.txt 是空的。
  
我也想问，试了好几组数据都无法触发pfc
  
You can tune the pfc threshold to trigger PFC in (switch_mmu.cc file)

**Comment by zqz-net** [comment]
> 

how about your email?

**Comment by william-learn** [comment]
> I have tried the (DCQCN + enable win), the PFC is not triggered. But only with **DCQCN,** the PFC is triggered. Is it reasonable?👀我已经尝试了 （DCQCN + enable win），未触发 PFC。但只有使用 DCQCN，才会触发 PFC。合理吗？ 👀
Hello, may I ask how you configured the experiment, and encountered PFC when running DCQCN?

**Comment by william-learn** [comment]
> > > > 我试过（DCQCN + enable win），PFC没有触发。但只有**DCQCN**才会触发 PFC。合理吗？👀我试过（DCQCN + enable win），PFC没有触发。但只有 DCQCN 才会触发 PFC。合理吗？👀
> > > 
> > > 
> > > 那么PFC触发时的测试用例是什么，测试的规模是多少？我已经尝试了 websearch 30%load 1 秒，并且只使用 DCQCN，但是 pfc.txt 是空的。那么PFC触发时的测试用例是什么，测试的规模是多少？ 我已经尝试了 websearch 30%load 1 秒，并且只使用 DCQCN，但是 pfc.txt 是空的。
> > 
> > 
> > 我也想问，试了好几组数据都无法触发pfc 我也想问，试了好几组数据都无法触发pfc
> 
> You can tune the pfc threshold to trigger PFC in (switch_mmu.cc file)您可以调整 pfc 阈值以触发 PFC（switch_mmu.cc 文件）

请问您是如何配置的实验，在运行dcqcn时出现了pfc

**Comment by ChaosPengs** [comment]
> > I have tried the (DCQCN + enable win), the PFC is not triggered. But only with **DCQCN,** the PFC is triggered. Is it reasonable?👀我已经尝试了 （DCQCN + enable win），未触发 PFC。但只有使用 DCQCN，才会触发 PFC。合理吗？ 👀
> > Hello, may I ask how you configured the experiment, and encountered PFC when running DCQCN?

python run.py --cc dcqcn --trace flow --bw 100 --topo topology
